### ISSUE TITLE
[Contact Journal] Add a button to the contact journal to add a self test

### ISSUE THREAD

**Issue by jucktnich** [corona-warn-app/cwa-wishlist#480]
## Current Implementation

Currently you can't add selftests to the contact journal.

## Suggested Enhancement

Make it possible to add a selftest to the contact journal with a timestamp.
## Expected Benefits

Let the Gesundheitsamt know, that you tested you negativ back than and you probably wasn't infective.

----
Internal Tracking-ID: [EXPOSUREAPP-6747](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-6747) obsolete
Internal Tracking-ID: [EXPOSUREAPP-4926](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-4926) story 
Internal Tracking-ID: [EXPOSUREAPP-5947](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-5947) iOS 
PR iOS corona-warn-app/cwa-app-ios#2749
Internal Tracking-ID: [EXPOSUREAPP-6008](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-6008) Android
PRs Android https://github.com/corona-warn-app/cwa-app-android/pull/3238
https://github.com/corona-warn-app/cwa-app-android/pull/3258

**Comment by fynngodau** [comment]
> Let the Gesundheitsamt know, that you tested you negativ back than and you probably wasn't infective.

I like your idea, however I'd like to note that to my knowledge, generally, self and rapid antigen tests only indicate an infection a few days *after* the infected person was already infective, and on the first day *after* symptom onset (PCR tests would be able to detect an infection much sooner). I've heard about that here: https://www.ndr.de/nachrichten/info/84-Nicht-auf-Tests-und-Impfungen-verlassen,audio869676.html

**Comment by jucktnich** [comment]
This may be so, but I think that selftests do have a 95% rate to be truly negative, so I hink the Gesundheitsamt can f.e. inform the contact persons, but not quarantine them.

**Comment by pkreissel** [comment]
I like the idea. The information could be used by the GA to prioritize contact tracing in times of scarce resources (like now). They know about the constraints of that information, so there is no harm in providing an easy way to add that. Also it could remind people to do regular rapid tests. It could even be a cool idea to add notification reminders about such tests to the app.

**Comment by heinezen** [comment]
Hey @jucktnich ,

Thank you for your request. We have added this issue to our internal Jira system (tracking ID: [EXPOSUREAPP-6747](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-6747)). The developers and the RKI will have a look and decide if and how the feature will be implemented. We will notify you on Github if we have more information from the development team that we can share.

----
Corona-Warn-App Open Source Team

**Comment by Ein-Tim** [comment]
If I interpret this PR right, this will be included in version 2.4: https://github.com/corona-warn-app/cwa-app-ios/pull/2749

Internal Tracking ID: EXPOSUREAPP-7310

**Comment by jucktnich** [comment]
Is this intended for approved tests (qr-code/tan) and/or for not approved tests (via a button for e.g. self tests)?

**Comment by Ein-Tim** [comment]
@jucktnich I assume only for tests that have been transferred via the in-app system to register tests, but that's only an assumption because it's way to early (for me) to tell from the code (although there are some hints like [this](https://github.com/corona-warn-app/cwa-app-ios/pull/2749/files#diff-f301e6a176659086bbf938a96ad16d9dc0394d6e0f793c110bd7162391d94347R497) which my assumption is based on). 

If my assumption is true your proposal wouldn't get implemented in the way you wanted it. But I think it still would be a nice enhancement.

**Comment by jucktnich** [comment]
Yeah would be still a nice enhancement, and the way to integrate other tests wouldn't be as long

**Comment by jucktnich** [comment]
@Ein-Tim what does this code say to you? I didn't found a hint for an assumption.

**Comment by Ein-Tim** [comment]
@jucktnich Like this line: https://github.com/corona-warn-app/cwa-app-ios/pull/2749/files#diff-f301e6a176659086bbf938a96ad16d9dc0394d6e0f793c110bd7162391d94347R497
It adds the CoronaTestType to the contact journal, "CoronaTestType" defines either PCR or RAT in the current implementation, for example in the data donation.

**Comment by jucktnich** [comment]
Ah yeah, but not limited too. Probably misunderstood your post above 😅

**Comment by dsarkar** [comment]
Also Android is in preparation: see
https://github.com/corona-warn-app/cwa-app-android/pull/3238
https://github.com/corona-warn-app/cwa-app-android/pull/3258

**Comment by heinezen** [comment]
Hello everyone,

This has been integrated into the current CWA release 2.4.

----
Corona-Warn-App Open Source Team

**Comment by jucktnich** [comment]
Not fully, cause you can't add selftests.

**Comment by Ein-Tim** [comment]
Since @jucktnich does not consider this as implemented, please remove the `Fix 2.4` label here.

**Comment by Ein-Tim** [comment]
Same here, please remove the `Fix 2.4` label from this issue. The author does not consider it as implemented.
Thank you!

cc @dsarkar & @larswmh
